<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>791</width>
    <height>572</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>:/img1/img/welcome.png</normaloff>:/img1/img/welcome.png</iconset>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonTextBesideIcon</enum>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>751</width>
      <height>541</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="page1">
     <widget class="QListWidget" name="listWidget_response">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>30</y>
        <width>721</width>
        <height>101</height>
       </rect>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_6">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>10</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="title">
       <string>动态</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page2">
     <widget class="QGroupBox" name="groupBox_book">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>741</width>
        <height>61</height>
       </rect>
      </property>
      <property name="title">
       <string>图书管理</string>
      </property>
      <widget class="QPushButton" name="pushButton_delete">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>400</x>
         <y>20</y>
         <width>75</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>更新图书</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/updata.png</normaloff>:/img/updata.png</iconset>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_booklist">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>75</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>图书列表</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/booklist.png</normaloff>:/img/booklist.png</iconset>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_addbook">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>270</x>
         <y>20</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>添加图书</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/addbook.png</normaloff>:/img/addbook.png</iconset>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_modifybook">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>20</y>
         <width>75</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>图书信息</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/modify.png</normaloff>:/img/modify.png</iconset>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>70</y>
        <width>741</width>
        <height>71</height>
       </rect>
      </property>
      <property name="title">
       <string>检索图书</string>
      </property>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>231</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>提示：可根据 书名 图书编号进行检索</string>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>231</width>
         <height>22</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>选择查询方式：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox_search"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>273</x>
         <y>20</y>
         <width>441</width>
         <height>26</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>输入图书信息查询：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_search">
          <property name="enabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>检索</string>
          </property>
          <property name="icon">
           <iconset resource="images.qrc">
            <normaloff>:/img/booksezrchicon.png</normaloff>:/img/booksezrchicon.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QTableView" name="tableView_book">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>731</width>
        <height>341</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page3">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>741</width>
        <height>61</height>
       </rect>
      </property>
      <property name="title">
       <string>读者管理</string>
      </property>
      <widget class="QPushButton" name="pushButton_readlist">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>75</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>读者列表</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/listreader.png</normaloff>:/img/listreader.png</iconset>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_radeadd">
       <property name="geometry">
        <rect>
         <x>340</x>
         <y>20</y>
         <width>75</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>添加读者</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/addreader.png</normaloff>:/img/addreader.png</iconset>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_readdel">
       <property name="geometry">
        <rect>
         <x>230</x>
         <y>20</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>删除读者</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/delreader.png</normaloff>:/img/delreader.png</iconset>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_4">
       <property name="geometry">
        <rect>
         <x>450</x>
         <y>20</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>更新读者</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/updata.png</normaloff>:/img/updata.png</iconset>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_modifyreader">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>20</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>基本信息</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/modify.png</normaloff>:/img/modify.png</iconset>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>70</y>
        <width>741</width>
        <height>81</height>
       </rect>
      </property>
      <property name="title">
       <string>检索读者</string>
      </property>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>211</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>提示：可根据 姓名 读者编号进行检索</string>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>221</width>
         <height>22</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>选择查询方式：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox_name"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>260</x>
         <y>30</y>
         <width>461</width>
         <height>26</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>输入读者信息查询：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_name"/>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_search">
          <property name="text">
           <string>检索</string>
          </property>
          <property name="icon">
           <iconset resource="images.qrc">
            <normaloff>:/img/booksezrchicon.png</normaloff>:/img/booksezrchicon.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QTableView" name="tableView_reader">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>160</y>
        <width>741</width>
        <height>321</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page4">
     <widget class="QGroupBox" name="groupBox_5">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>741</width>
        <height>71</height>
       </rect>
      </property>
      <property name="title">
       <string>借阅管理</string>
      </property>
      <widget class="QPushButton" name="pushButton_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>75</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>借阅记录</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/recordana.png</normaloff>:/img/recordana.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_6">
       <property name="geometry">
        <rect>
         <x>330</x>
         <y>30</y>
         <width>75</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>借阅书籍</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/record.png</normaloff>:/img/record.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_deleterecord">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>30</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>归还书籍</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/bookdelepng.png</normaloff>:/img/bookdelepng.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_continue">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>30</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>续借图书</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/xujie.png</normaloff>:/img/xujie.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
      </widget>
     </widget>
     <widget class="QTableView" name="tableView_record">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>80</y>
        <width>741</width>
        <height>411</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page5">
     <widget class="QGroupBox" name="groupBox_4">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>741</width>
        <height>71</height>
       </rect>
      </property>
      <property name="title">
       <string>系统操作</string>
      </property>
      <widget class="QPushButton" name="pushButton_datacopy">
       <property name="geometry">
        <rect>
         <x>240</x>
         <y>30</y>
         <width>75</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>数据备份</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/backup.png</normaloff>:/img/backup.png</iconset>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_changepwd">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>30</y>
         <width>75</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>用户退出</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/loginicon.png</normaloff>:/img/loginicon.png</iconset>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton__perinfo">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>75</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="text">
        <string>用户类型</string>
       </property>
       <property name="icon">
        <iconset resource="images.qrc">
         <normaloff>:/img/infor.png</normaloff>:/img/infor.png</iconset>
       </property>
      </widget>
      <widget class="QLineEdit" name="hostLineEdit">
       <property name="geometry">
        <rect>
         <x>540</x>
         <y>10</y>
         <width>113</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="portLineEdit">
       <property name="geometry">
        <rect>
         <x>540</x>
         <y>40</y>
         <width>113</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>500</x>
         <y>10</y>
         <width>54</width>
         <height>12</height>
        </rect>
       </property>
       <property name="text">
        <string>IP：</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>500</x>
         <y>40</y>
         <width>54</width>
         <height>12</height>
        </rect>
       </property>
       <property name="text">
        <string>Port：</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_listen">
       <property name="geometry">
        <rect>
         <x>660</x>
         <y>10</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>连接服务</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_close">
       <property name="geometry">
        <rect>
         <x>660</x>
         <y>40</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>关闭服务</string>
       </property>
      </widget>
     </widget>
     <widget class="QLabel" name="label_connect">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>490</y>
        <width>311</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>客户端端状态</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="label_name">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>10</y>
      <width>191</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>当前用户信息</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_time">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>10</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>系统时间</string>
    </property>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="mouseTracking">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonFollowStyle</enum>
   </property>
   <property name="floatable">
    <bool>true</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="action"/>
   <addaction name="BookAction"/>
   <addaction name="Readaction"/>
   <addaction name="BorrowAction"/>
   <addaction name="Systemaction"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>791</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>图书信息管理</string>
    </property>
    <addaction name="action_3"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>读者信息管理</string>
    </property>
    <addaction name="action_4"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>图书借阅管理</string>
    </property>
    <addaction name="action_5"/>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>系统管理</string>
    </property>
    <addaction name="action_6"/>
   </widget>
   <widget class="QMenu" name="menu_5">
    <property name="title">
     <string>&amp;首页</string>
    </property>
    <addaction name="action_2"/>
   </widget>
   <widget class="QMenu" name="menu_6">
    <property name="title">
     <string>关于</string>
    </property>
    <addaction name="action_8"/>
    <addaction name="action_7"/>
   </widget>
   <addaction name="menu_5"/>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
   <addaction name="menu_4"/>
   <addaction name="menu_6"/>
  </widget>
  <action name="BookAction">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/img/book.png</normaloff>:/img/book.png</iconset>
   </property>
   <property name="text">
    <string>图书信息管理</string>
   </property>
   <property name="toolTip">
    <string>图书信息管理</string>
   </property>
  </action>
  <action name="Readaction">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/img/reader.png</normaloff>:/img/reader.png</iconset>
   </property>
   <property name="text">
    <string>读者信息管理</string>
   </property>
   <property name="toolTip">
    <string>读者信息管理</string>
   </property>
  </action>
  <action name="Systemaction">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/img/set.png</normaloff>:/img/set.png</iconset>
   </property>
   <property name="text">
    <string>系统管理</string>
   </property>
  </action>
  <action name="action">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/img/welcome.png</normaloff>:/img/welcome.png</iconset>
   </property>
   <property name="text">
    <string>首页</string>
   </property>
   <property name="toolTip">
    <string>首页</string>
   </property>
  </action>
  <action name="BorrowAction">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/img/recordaction.png</normaloff>:/img/recordaction.png</iconset>
   </property>
   <property name="text">
    <string>图书借阅管理</string>
   </property>
  </action>
  <action name="action_2">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/img/welcome.png</normaloff>:/img/welcome.png</iconset>
   </property>
   <property name="text">
    <string>欢迎</string>
   </property>
  </action>
  <action name="action_3">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/img/book.png</normaloff>:/img/book.png</iconset>
   </property>
   <property name="text">
    <string>图书区</string>
   </property>
  </action>
  <action name="action_4">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/img/reader.png</normaloff>:/img/reader.png</iconset>
   </property>
   <property name="text">
    <string>读者区</string>
   </property>
  </action>
  <action name="action_5">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/img/recordaction.png</normaloff>:/img/recordaction.png</iconset>
   </property>
   <property name="text">
    <string>借阅区</string>
   </property>
  </action>
  <action name="action_6">
   <property name="icon">
    <iconset resource="images.qrc">
     <normaloff>:/img/set.png</normaloff>:/img/set.png</iconset>
   </property>
   <property name="text">
    <string>设置</string>
   </property>
  </action>
  <action name="action_8">
   <property name="text">
    <string>作者</string>
   </property>
  </action>
  <action name="action_7">
   <property name="text">
    <string>版本</string>
   </property>
   <property name="toolTip">
    <string>版本</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>action_2</sender>
   <signal>triggered()</signal>
   <receiver>action</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_6</sender>
   <signal>triggered()</signal>
   <receiver>Systemaction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_4</sender>
   <signal>triggered()</signal>
   <receiver>Readaction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_3</sender>
   <signal>triggered()</signal>
   <receiver>BookAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_5</sender>
   <signal>triggered()</signal>
   <receiver>BorrowAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
